rules/DowngradePhp70/Rector/GroupUse/SplitGroupedUseImportsRector.php

```diff
@@ -53,6 +58,20 @@ public function getNodeTypes(): array
      */
     public function refactor(Node $node): array
     {
+        $uses = $this->useImportsResolver->resolveBareUsesForNode($node);
+
+        /**
+         * when combined with Use_, it got duplicated ;
+         * so ; need to be removed early
+         */
+        if ($uses !== []) {
+            $oldTokens = $this->file->getOldTokens();
+            $endTokenPost = $node->getEndTokenPos();
+            unset($oldTokens[$endTokenPost + 1]);
+            $oldTokens = array_values($oldTokens);
+            $this->file->setOldTokens($oldTokens);
+        }
```

Comment:
There must be some better way to deal with this.
Maybe it's a bug in php-parser that should be fixed there.

What happens when you try the other lexer in https://github.com/TomasVotruba/group-use-test ?
If it fails, the issue is in php-parser and should be reported there.

Reply:
as I commented at https://github.com/rectorphp/rector-src/pull/2578#issuecomment-1167530485

It seems we use `PhpParser\Lexer` instead of `PhpParser\Lexer\Emulative` 

https://github.com/rectorphp/rector-src/blob/e81691f1c68a976c1589830642cd45873389d962/config/phpstan/parser.neon#L27

changing it in factory:

```php
    public function createEmulativeLexer(): Lexer
    {
        return new \PhpParser\Lexer\Emulative([
            'usedAttributes' => [
                'comments',
                'startLine', 'endLine',
                'startTokenPos', 'endTokenPos',
            ],
        ]);
    }
```

will got error:

```bash
There was 1 error:

1) Rector\Tests\DowngradePhp70\Rector\GroupUse\SplitGroupedUseImportsRector\SplitGroupedUseImportsRectorTest::test with data set #0 (Symplify\SmartFileSystem\SmartFileInfo Object (...))
TypeError: PhpParser\Lexer::getTokens(): Return value must be of type array, null returned
```